pkg/util/wsstream/conn.go

```diff
@@ -123,21 +124,28 @@ func handshake(config *websocket.Config, req *http.Request, allowed []string) er
 // Conn supports sending multiple binary channels over a websocket connection.
 // Supports only the "channel.k8s.io" subprotocol.
 type Conn struct {
-	channels []*websocketChannel
-	codec    codecType
-	ready    chan struct{}
-	ws       *websocket.Conn
-	timeout  time.Duration
+	channels           []*websocketChannel
+	codec              codecType
+	ready              chan struct{}
+	ws                 *websocket.Conn
+	timeout            time.Duration
+	supportedProtocols []string
+	protocolVersion    string
 }
 
 // NewConn creates a WebSocket connection that supports a set of channels. Channels begin each
 // web socket message with a single byte indicating the channel number (0-N). 255 is reserved for
 // future use. The channel types for each channel are passed as an array, supporting the different
 // duplex modes. Read and Write refer to whether the channel can be used as a Reader or Writer.
-func NewConn(channels ...ChannelType) *Conn {
+//
+// Protocols have the shape "channel.k8s.io" in the unversioned case and "channel.k8s.io/v4" in
```

Comment:
It's not simpler but it's consistent with the decision we made on protocols
previously.  I don't want to invent a new mixing scheme.  Each protocol is
independent.

On Fri, Aug 5, 2016 at 1:18 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In pkg/util/wsstream/conn.go
> <https://github.com/kubernetes/kubernetes/pull/26541#discussion_r73643040>
> :
>
> >  }
> >
> >  // NewConn creates a WebSocket connection that supports a set of channels. Channels begin each
> >  // web socket message with a single byte indicating the channel number (0-N). 255 is reserved for
> >  // future use. The channel types for each channel are passed as an array, supporting the different
> >  // duplex modes. Read and Write refer to whether the channel can be used as a Reader or Writer.
> > -func NewConn(channels ...ChannelType) *Conn {
> > +//
> > +// Protocols have the shape "channel.k8s.io" in the unversioned case and "channel.k8s.io/v4" in
>
> it's less about being fancy, but about easy distinction what is handled
> inside the Conn (binary vs. b64) and what is provided from outside. Using
> the version as prefix with a dot is not simpler.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26541/files/ac7867dd7be0fdf3a417cc7fb69cb95cfccb12b0..176f97d29edfe3228ce86ffddc957d7146c1e3bd#r73643040>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pxxftQZ201lmO0PzuYbbXHKp0j6kks5qcseXgaJpZM4Ip8Gm>
> .
>